api/src/services/__tests__/programsService.ts

```diff
@@ -0,0 +1,372 @@
+      const programId = 2;
+      const curriculumId = 2;
+
+      const [, matchingProgram] = programsList;
+
+      const matchingCurriculumActivities: CurriculumActivity[] = [
+        curriculumActivitiesList[3],
+        curriculumActivitiesList[4],
+        curriculumActivitiesList[5],
+      ];
+
+      const matchingProgramActivities: ProgramActivity[] = [
+        program2ActivitiesList[0],
+        program2ActivitiesList[1],
+        program2ActivitiesList[2],
+      ];
+
+      // findProgram
+      mockQuery(
+        'select `id`, `title`, `start_date`, `end_date`, `time_zone`, `curriculum_id` from `programs` where `id` = ?',
+        [programId],
+        [matchingProgram]
+      );
+
+      // listCurriculumActivities
+      mockQuery(
+        'select `id`, `title`, `description_text`, `curriculum_week`, `curriculum_day`, `start_time`, `end_time`, `duration`, `activity_type_id`, `curriculum_id` from `activities` where `curriculum_id` = ?',
+        [curriculumId],
+        matchingCurriculumActivities
+      );
+
+      // activity types
+      mockQuery(
+        'select `id`, `title` from `activity_types`',
+        [],
+        activityTypesList
+      );
+
+      // listProgramActivities
+      expect(await listProgramActivities(programId)).toEqual(
+        matchingProgramActivities
+      );
+    });
+  });
+
+  describe('findProgramWithActivities', () => {
+    it('should give the details of the specified program including its activities', async () => {
+      const programId = 2;
+
+      const [, matchingProgram] = programsList;
+
+      const matchingCurriculumActivities: CurriculumActivity[] = [
+        curriculumActivitiesList[3],
+        curriculumActivitiesList[4],
+        curriculumActivitiesList[5],
+      ];
+
+      const matchingProgramActivities: ProgramActivity[] = [
+        program2ActivitiesList[0],
+        program2ActivitiesList[1],
+        program2ActivitiesList[2],
+      ];
+
+      const programWithActivities = JSON.parse(JSON.stringify(matchingProgram));
+      programWithActivities.activities = matchingProgramActivities;
+
+      // findProgram
+      mockQuery(
+        'select `id`, `title`, `start_date`, `end_date`, `time_zone`, `curriculum_id` from `programs` where `id` = ?',
+        [programId],
+        [matchingProgram]
+      );
+
+      //listCurriculumActivities
+      mockQuery(
+        'select `id`, `title`, `description_text`, `curriculum_week`, `curriculum_day`, `start_time`, `end_time`, `duration`, `activity_type_id`, `curriculum_id` from `activities` where `curriculum_id` = ?',
+        [matchingProgram.curriculum_id],
+        matchingCurriculumActivities
+      );
+
+      // activityTypes
+      mockQuery(
+        'select `id`, `title` from `activity_types`',
+        [],
+        activityTypesList
+      );
+
+      expect(await findProgramWithActivities(programId)).toEqual(
+        programWithActivities
+      );
+    });
+  });
+});
```

Comment:
Overall whitespace can be cleaned up on programsService.ts file.

Reply:
I tightened up all whitespace by removing all line breaks as well as adjoining whitespace characters, then ran a `yarn delint`. Ultimately, we're at the mercy of our `prettier` settings here. I welcome any and all feedback resilient to the `yarn delint` process that you think may help us with tightening up whitespace.